src/openms/source/CHEMISTRY/Tagger.cpp

```diff
@@ -19,36 +19,42 @@
 
 namespace OpenMS
 {
-  char Tagger::getAAByMass_(double m) const
+  // Define a method to get amino acid by mass
+char Tagger::getAAByMass_(double m) const
   {
     // fast check for border cases
-    if (m < min_gap_ || m > max_gap_) return ' ';
-
+    if (m < min_gap_ || m > max_gap_) return ' '; // If mass is out of the valid range, return a space character
+    // Calculate delta using a function from the Math class, converting ppm to mass
     const double delta = Math::ppmToMass(ppm_, m);
+    // Find the lower bound of m - delta in mass2aa
     auto left = mass2aa_.lower_bound(m - delta);
     //if (left == mass2aa_.end()) return ' '; // cannot happen, since we checked boundaries above
-
-    if (fabs(left->first - m) < delta) return left->second;
+  // Check if the absolute difference between left->first and m is smaller than delta
+    if (fabs(left->first - m) < delta) return left->second; // If yes, return the associated amino acid
     return ' ';
   }
 
   void Tagger::getTag_(std::string & tag, const std::vector<double>& mzs, const size_t i, std::vector<std::string>& tags, const size_t charge) const
   {
     const size_t N = mzs.size();
     size_t j = i + 1;
+    // Iterate through the mzs vector starting from index j
     while (j < N)
     {
+      // Check if the tag size has reached the maximum allowed length
```

Comment:
basically what is written in the line below. maybe some higher-level comment here?